### PR TITLE
ref(*): add/use resource consts

### PR DESCRIPTION
**Pull request by vdice**
**What this PR does / why we need it**:

* Adds consts for commonly used resource names and updates applicable logic to use them

Closes https://github.com/brigadecore/brigade/issues/1174

**Special notes for your reviewer**:

There are still some resources not const-ified (search the repo for all instances of `Kind:` in creating meta.TypeMeta structs)... do we want to update these as well, even though there currently might only be one use of such strings?

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

**Review by radu-matei — APPROVED** [review]

File: sdk/v2/authn/users.go [hunk 1]
```diff
@@ -12,6 +12,9 @@ import (
 	"github.com/brigadecore/brigade/sdk/v2/restmachinery"
 )
 
+// UserLabel represents the canonical User label string
+const UserLabel = "User"
```
Comment (krancour) [inline sdk/v2/authn/users.go:16]: Based on how this is being used, would `UserKind` be more appropriate?

Comment (krancour) [reply sdk/v2/authn/users.go:16]: Same comment applies in several other places.

Comment (vdice) [reply sdk/v2/authn/users.go:16]: Sure, shall I update all of these to `*Kind`?

Comment (krancour) [reply sdk/v2/authn/users.go:16]: Yes... unless you disagree that my suggestion adds clarity to how those constants are used.

**Review by krancour — APPROVED** [review]

**Comment by krancour** [comment]
LGTM. Integration tests failed... re-running.

**Pull request merged**